common/lib/xmodule/xmodule/x_module.py

```diff
@@ -766,7 +768,18 @@ def public_view(self, _context):
             u'<span class="icon icon-alert fa fa fa-warning" aria-hidden="true"></span>'
             u'<div class="message-content">{}</div></div></div>'
         )
-        return Fragment(alert_html.format(DEFAULT_PUBLIC_VIEW_MESSAGE))
+
+        if self.display_name:
+            display_text = _(
+                "{display_name} is only accessible to enrolled learners."
```

Comment:
This looks like it the message is a byte string and might run into encoding issues. Have you tested what happens if `self.display_name` is non-ASCII unicode text?

Reply:
The `path` value used for the sequence does seems to be a byte-string as well? https://github.com/edx/edx-platform/blob/6f0e5d68d7b75fc954703582700bd5aa6e971f40/common/lib/xmodule/xmodule/seq_module.py#L494 

@pkulkark Can you please add a test for this in any case?

Reply:
I've corrected it to use unicode instead of byte-string. 
@symbolist I wasn't able to find a suitable place for a test case, can you please let me know where I can add one?

Reply:
I think this TestCase class should work: https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/tests/test_xblock_wrappers.py#L322